[PT][FSDP] Combine _utils.py into _common_utils.py [1/3]

AlbertDachiChen opened the pull request:
Summary:
https://github.com/pytorch/pytorch/issues/97813

This diffs moves `_override_module_mixed_precision`

Test Plan: CI

Differential Revision: D47706059

awgu reviewed (approved):
LGTM! Thanks for the quick implementation!

AlbertDachiChen committed:
[PT][FSDP] Combine _utils.py into _common_utils.py [1/3] (#105857)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105857

https://github.com/pytorch/pytorch/issues/97813

This diffs moves `_override_module_mixed_precision`

Test Plan: CI

Reviewed By: awgu

Differential Revision: D47706059

fbshipit-source-id: 01103f7df87635c123b5aeef3e4e610da303fe31

awgu commented:
Failure in 
pull / linux-bionic-cuda12.1-py3.10-gcc9-sm86 / test (default, 1, 5, linux.g5.4xlarge.nvidia.gpu) is unrelated:
```
FAILED [0.0136s] test_transformers.py::TestSDPACUDA::test_mem_efficient_attention_vs_math_ref_grads_batch_size_8_seq_len_q_2048_seq_len_k_2048_head_dim_32_is_causal_False_dropout_p_0_0_float32_scale_None_cuda_float32 - torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 16.00 MiB. GPU 0 has a total capacty of 21.99 GiB of which 12.58 GiB is free. Process 7616 has 248.00 MiB memory in use. Process 12405 has 8.90 GiB memory in use. Process 12516 has 248.00 MiB memory in use. Of the allocated memory 3.10 GiB is allocated by PyTorch, and 5.46 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

To execute this test, run the following from the base repo dir:
     python test/test_transformers.py -k test_mem_efficient_attention_vs_math_ref_grads_batch_size_8_seq_len_q_2048_seq_len_k_2048_head_dim_32_is_causal_False_dropout_p_0_0_float32_scale_None_cuda_float32
```

awgu commented:
@pytorchbot merge -f "unrelated failure"
